source/mainnet/smart-contracts/general/introduction.rst

```diff
@@ -88,6 +88,22 @@ module <contract-module>`. After this a smart contract can be *initialized* to
 obtain a :ref:`smart contract instance <contract-instances>`. Finally a smart
 contract instance can be repeatedly updated according to its own logic.
 
+.. image:: images/smart-contract-lifecycle.png
+    :width: 100%
+    :alt: flow diagram with different actions
+
+#. In ``cargo-concordium`` :ref:`run the init command<setup-contract>` to start a new project.
+
+#. Edit your contract, including the entrypoints, functions, and parameters necessary to execute what is needed. If :ref:`using a schema<build-schema>`, make sure that the contract is prepared for this.
+
+#. In ``cargo-concordium`` :ref:`run the build command<compile-module>` to build the Wasm module that can be deployed on chain.
+
+#. In ``concordium-client`` :ref:`run the deploy command<deploy-module>` to deploy the Wasm module. This makes the contract available on chain.
+
+#. In ``concordium-client`` :ref:`run the init command<initialize-contract>` to initialize the contract on chain. This gives you a fresh state.
+
+#. In ``concordium-client`` you can then :ref:`run invoke<invoke-instance>` to simulate or test your contract and see what kind of energy it uses, :ref:`contract show<inspect-instance>` to see the schema or parameters in the contract, or :ref:`update<interact-instance>` to execute transactions and update the state.
```

Comment:
Nitpick: Maybe the box "Contract show" on the diagram should be named just "Show" for consistency? It seems the other blocks are commands, that follow after the prefix `concordium-client contract` (and then `invoke`/`update`/`show`)

Reply:
The "Contract show" box on the diagram is connected with an arrow to another arrow, the one between "Initialize" and "Update". Why it's not connecting Initialize and "Contract show" directly?

There is also a two-headed arrow between "Invoke" and "Update". What does it show?

Reply:
> what kind of energy it uses,

Maybe "how much energy" or something like that.

Reply:
> to simulate or test your contract

Both simulate and test sound similar to me. A different use case of `invoke` is to call a view entrypoint, that is, an entrypoint that returns some data derived from the contract state.

Reply:
Maybe the two headed arrow between invoke and update should only go one way - from invoke to update. You would not invoke again after you have run an update...or would you to view an entrypoint?

Reply:
I think one can use invoke for a dry-run, then update and then invoke a view, that returns the contract state after update. So, I guess it's fine to keep it both ways.